openedx/core/djangoapps/programs/utils.py

```diff
@@ -424,3 +432,31 @@ def _attach_run_mode_upgrade_url(self, run_mode):
                 run_mode['upgrade_url'] = None
         else:
             run_mode['upgrade_url'] = None
+
+
+def course_start_datetime_text(start_date, advertised_start=None):
```

Comment:
Why not have this take a course object instead?  And then callers don't need to worry about passing in the dates individually?